syscalls_linux.go

```diff
@@ -31,50 +31,52 @@ func ioctl(fd uintptr, request uintptr, argp uintptr) error {
 }
 
 func newTAP(config Config) (ifce *Interface, err error) {
-	file, err := os.OpenFile("/dev/net/tun", os.O_RDWR, 0)
+	fdInt, err := syscall.Open("/dev/net/tun", os.O_RDWR|syscall.O_NONBLOCK, 0)
 	if err != nil {
 		return nil, err
 	}
+	fd := uintptr(fdInt)
 
 	var flags uint16
 	flags = cIFF_TAP | cIFF_NO_PI
 	if config.PlatformSpecificParams.MultiQueue {
 		flags |= cIFF_MULTI_QUEUE
 	}
-	name, err := createInterface(file.Fd(), config.Name, flags)
+	name, err := createInterface(fd, config.Name, flags)
 	if err != nil {
 		return nil, err
 	}
 
-	if err = setDeviceOptions(file.Fd(), config); err != nil {
+	if err = setDeviceOptions(fd, config); err != nil {
 		return nil, err
 	}
 
-	ifce = &Interface{isTAP: true, ReadWriteCloser: file, name: name}
+	ifce = &Interface{isTAP: true, ReadWriteCloser: os.NewFile(fd, "tun"), name: name}
```

Comment:
Name should be `"tap"` instead of `"tun"`.

Reply:
Oh this is actually `tun` I think. Although this is a tap interface, it's still from `/dev/net/tun`